### ISSUE TITLE
Paging in `HistoryView` of `ModelViewSet`/`SnippetViewSet` does not work

### ISSUE THREAD

**Issue by th3hamm0r** [wagtail/wagtail#11569]
### Issue Summary

While implementing a custom listing/index in a `ModelViewSet` (actually a `SnippetViewSet`), which is based on the code of the `HistoryView` ([code](https://github.com/wagtail/wagtail/blob/c221a802cac7234f2541c91868956ccf31af10cd/wagtail/admin/views/generic/history.py#L62)), I've found out, that the paging does not work here.
We're using wagtail 5.2.x, but I've reproduced it with 6.0 too.

The paginator uses the `index_url`, which is set [here](https://github.com/wagtail/wagtail/blob/c221a802cac7234f2541c91868956ccf31af10cd/wagtail/admin/views/generic/base.py#L414), where the value comes from [here](https://github.com/wagtail/wagtail/blob/c221a802cac7234f2541c91868956ccf31af10cd/wagtail/admin/views/generic/base.py#L393-L399), where the `index_url_name` gets passed for all default views by the `ModelViewSet` ([here](https://github.com/wagtail/wagtail/blob/c221a802cac7234f2541c91868956ccf31af10cd/wagtail/admin/viewsets/model.py#L126)).

**Solution:**
Based on how we've solved it for our custom view, I think we could fix it by adding the following to the generic `HistoryView` ([code](https://github.com/wagtail/wagtail/blob/c221a802cac7234f2541c91868956ccf31af10cd/wagtail/admin/views/generic/history.py#L62)):
```python
history_url_name = None
...
def get_index_url(self):
    if self.history_url_name:
        return reverse(self.history_url_name, args=(self.pk,))
```
This adds the `history_url_name` so it gets passed by `construct_view()` and overrides the default `get_index_url()`.

Note: This seems to work, but I've only tested it quickly in an actual project.

### Steps to Reproduce

1. Install the [bakerydemo](https://github.com/wagtail/bakerydemo)
2. Login to the CMS
3. Go to "Bakery Misc" -> "People" and open one person (e.g. http://127.0.0.1:8000/admin/snippets/base/person/edit/4/).
4. Click "Save draft" often enough, that the history view (e.g. http://127.0.0.1:8000/admin/snippets/base/person/history/4/) needs paging (= 20+1 items) ;)
5. Using the paginator to get to the next page should result in an 404

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.11
-   Django version: 4.2.9
-   Wagtail version: 5.2.3, 6.0rc1, 6.0 (latest `stable/6.0.x`)

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by abdlrhman08** [comment]
I reproduced the bug and it does happen. I'll give this one a try

**Comment by abdlrhman08** [comment]
Omitting the `{{ url path }}` in the hyperling tag in pagination_nav.html currently fixes it permanently, since the queries are always done on the same url. I don't know if it can be considered "a good fix" since it will apply on all views that depend on pagination_nav.html . [This is the line I am talking about](https://github.com/wagtail/wagtail/blob/e15866663c709533939f8073d98894aaa0095012/wagtail/admin/templates/wagtailadmin/shared/pagination_nav.html#L16)

**Comment by th3hamm0r** [comment]
At least your solution would avoid, that with every new subclass of `BaseListingView` added to the `ModelViewSet` you would have to pass the view's url itself to the view (as with my solution, which is not very DRY).
Alternatively, maybe with less side-effects: remove the `linkurl=index_url` from [listing_results.html](https://github.com/wagtail/wagtail/blob/e15866663c709533939f8073d98894aaa0095012/wagtail/admin/templates/wagtailadmin/generic/listing_results.html#L11).

There are multiple cases where `linkurl` is set (which results in `{{ url_path }}`) is used, so I'm also not sure, if we could omit it completely.

**Comment by gasman** [comment]
The same issue also affects the Usage view.

**Comment by gasman** [comment]
> `index_url_name = self.get_url_name("history")`

Actually, this doesn't work, since BaseListingView expects to be able to reverse this with no arguments - @th3hamm0r's original suggestion is the approach to go for after all 👍

**Comment by laymonage** [comment]
One thing to consider is that once the `get_index_url()` method has been overridden to use the `history_url_name`, the `get_breadcrumbs_items()` method must be updated to `reverse()` the URL to the index view instead of relying on the `get_index_url()` method.

**Comment by gasman** [comment]
@laymonage Just noticed that myself, but I think we're fine because `get_breadcrumbs_items` (on both views) is already calling `reverse(self.index_url_name)` rather than going through `get_index_url`. The fact that these views are using 'index' for the actual index view as well as for 'self' is all the more reason for us to fix this properly for 6.1 though :-)
